File: Java/src/com/ouroboros/include/Apple.java
package com.ouroboros.include;

import java.util.Random;

public final class Apple {
    public int x, y;
    private final Random random = new Random();

    public Apple(Snake snake) {
        eat(snake);
    }

    public void eat(Snake snake) {
        for (int i = 0; i < 1000; ++i) {
            boolean blocked = false;
            int rx = random.nextInt(20);
            int ry = random.nextInt(20);

            for (Node node : snake.head) {
                if (node.x == rx && node.y == ry) {
                    blocked = true;
                    break;
                }
            }

            if (!blocked) {
                this.x = rx;
                this.y = ry;
                return;
            }
        }
    }
}

File: Kotlin/app/src/main/kotlin/org/example/apple.kt
package org.example

import kotlin.random.Random

class Apple(snake: Snake) {
    var x: Int = 0
    var y: Int = 0

    init {
        eat(snake)
    }

    public fun eat(snake: Snake) {
        repeat(1000) {
            var blocked = false
            val rx = Random.nextInt(0, 20)
            val ry = Random.nextInt(0, 20)

            for (node in snake.head) {
                if (node.x == rx && node.y == ry) {
                    blocked = true
                    break
                }
            }

            if (!blocked) {
                x = rx
                y = ry
                return
            }
        }
    }
}